snapcraft/plugins/go.py

```diff
@@ -111,7 +112,7 @@ def pull(self):
         super().pull()
         os.makedirs(self._gopath_src, exist_ok=True)
 
-        if self.options.source:
+        if glob('{}/**/*.go'.format(self.sourcedir), recursive=True):
```

Comment:
because it may sometimes not be empty and confuse the `go` plugin. Check the webcam-webui demo and see